units/recalculate_change_points_test.go

```diff
@@ -0,0 +1,213 @@
+	"os"
+	"runtime"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/evergreen-ci/cedar/perf"
+
+	"github.com/evergreen-ci/cedar"
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/mgo.v2/bson"
+)
+
+type TestResultsAndRollups struct {
+	info    *model.PerformanceResultInfo
+	rollups []model.PerfRollupValue
+}
+
+func makePerfResults(length int, breakpoint int, unique string) []TestResultsAndRollups {
+	var rollups []TestResultsAndRollups
+	i := 0
+	for i < length {
+		newRollup := TestResultsAndRollups{
+			info: &model.PerformanceResultInfo{
+				Project:  "project" + unique,
+				Variant:  "variant",
+				Version:  "version" + strconv.Itoa(i+1),
+				Order:    i + 1,
+				TestName: "test",
+				TaskName: "task",
+				Mainline: true,
+			},
+			rollups: []model.PerfRollupValue{
+				{
+					Name:       "measurement",
+					MetricType: model.MetricTypeSum,
+					Version:    0,
+				},
+				{
+					Name:       "measurement_another",
+					MetricType: model.MetricTypeSum,
+					Version:    0,
+				},
+			},
+		}
+		if i < breakpoint {
+			newRollup.rollups[0].Value = float64(100)
+			newRollup.rollups[1].Value = float64(100)
+		} else {
+			newRollup.rollups[0].Value = float64(1000)
+			newRollup.rollups[1].Value = float64(1000)
+		}
+		rollups = append(rollups, newRollup)
+		i++
+	}
+	return rollups
+}
+
+func init() {
+	dbName := "cedar_signal_processing"
+	env, err := cedar.NewEnvironment(context.Background(), dbName, &cedar.Configuration{
+		MongoDBURI:    "mongodb://localhost:27017",
+		DatabaseName:  dbName,
+		SocketTimeout: time.Minute,
+		NumWorkers:    2,
+	})
+	if err != nil {
+		panic(err)
+	}
+	cedar.SetEnvironment(env)
+}
+
+func tearDown(env cedar.Environment) error {
+	conf, session, err := cedar.GetSessionWithConfig(env)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer session.Close()
+	return errors.WithStack(session.DB(conf.DatabaseName).DropDatabase())
+}
+
+type MockDetector struct {
+	Calls [][]float64
+}
+
+func (m *MockDetector) DetectChanges(ctx context.Context, series []float64) ([]perf.ChangePoint, error) {
+	m.Calls = append(m.Calls, series)
+	last := series[0]
+	var cps []int
+	for idx, i := range series {
+		if i != last {
+			last = i
+			cps = append(cps, idx)
+		}
+	}
+	var changePoints []perf.ChangePoint
+	for _, cp := range cps {
+		changePoints = append(changePoints, perf.ChangePoint{
+			Index: cp,
+			Algorithm: model.AlgorithmInfo{
+				Name:    "some_algorithm",
+				Version: 1,
+				Options: []model.AlgorithmOption{
+					{
+						Name:  "some_option",
+						Value: 5,
+					},
+					{
+						Name:  "another_option",
+						Value: 0.05,
+					},
+				},
+			},
+		})
+	}
+
+	return changePoints, nil
+}
+
+func TestRecalculateChangePointsJob(t *testing.T) {
+
+	if runtime.GOOS == "darwin" && os.Getenv("EVR_TASK_ID") != "" {
+		t.Skip("avoid less relevant failing test in evergreen")
+	}
```

Comment:
I mean this makes the test basically useless, and means that I'll never run your tests?

Reply:
Copied from `TestFTDCRollupsJob`. Will remove here, but I guess that test is useless as well then?